pkg/interface/src/views/apps/profile/components/EditProfile.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import * as Yup from "yup";
+
+import {
+  ManagedForm as Form,
+  ManagedTextInputField as Input,
+  ManagedCheckboxField as Checkbox,
+  Center,
+  Col,
+  Box,
+  Text,
+  Row,
+  Button,
+} from "@tlon/indigo-react";
+import { Formik, FormikHelpers } from "formik";
+import { useHistory } from "react-router-dom";
+
+import { uxToHex } from "~/logic/lib/util";
+import { Sigil } from "~/logic/lib/sigil";
+import { AsyncButton } from "~/views/components/AsyncButton";
+import { ColorInput } from "~/views/components/ColorInput";
+import { ImageInput } from "~/views/components/ImageInput";
+import { MarkdownField } from "~/views/apps/publish/components/MarkdownField";
+import { resourceFromPath } from "~/logic/lib/group";
+import GroupSearch from "~/views/components/GroupSearch";
+
+
+const formSchema = Yup.object({
+  nickname: Yup.string(),
+  bio: Yup.string(),
+  color: Yup.string(),
+  avatar: Yup.string().nullable()
+});
+
+const emptyContact = {
+  nickname: '',
+  bio: '',
+  status: '',
+  color: '0',
+  avatar: null,
+  cover: null,
+  groups: [],
+  'last-updated': 0,
+  isPublic: false
+};
+
+
+export function EditProfile(props: any) {
+  const { contact, ship, api, isPublic } = props;
+  const history = useHistory();
+  if (contact) {
+    contact.isPublic = isPublic;
+  }
+
+  const onSubmit = async (values: any, actions: any) => {
+    console.log(values);
+    try {
+      await Object.keys(values).reduce((acc, key) => {
+        console.log(key);
+        const newValue = key !== "color" ? values[key] : uxToHex(values[key]);
```

Comment:
Is it necessary to make these sequential? I'd prefer a parallel promise.all considering there's no implied order to the edits?